### PR TITLE
feat: add collapsible theme

### PR DESCRIPTION
**Pull request by asanehisa**
Screenshot 2024-11-18 at 5 40 38 PM

### CODE REVIEW

**Review by benlife5 — COMMENTED** [review]
Nice! Two styling nits:
1. Try to get the spacing around the border of warning at the top to match the spacing around the border of the theme sections 
2. Might need to consult with Aaron/design on this but the top level FieldLabels don't really look like what I'd expect for a dropdown heading/trigger (I created them, sorry about that haha 😬)

File: src/internal/puck/components/ThemeSidebar.tsx [hunk 1]
```diff
@@ -59,21 +77,33 @@ const ThemeSidebar = (props: ThemeSidebarProps) => {
...
-              }
-              value={values}
-            />
-          </FieldLabel>
+          <div key={themeSectionKey} className="theme-field">
+            <button
+              className="flex items-center gap-0.5"
+              onClick={() => toggleSection(themeSectionKey)}
+            >
+              {isCollapsed ? `${field.label ?? ""}` : `${field.label ?? ""}`}
```
[the remark this answers is not in the document]
Comment (asanehisa) [reply src/internal/puck/components/ThemeSidebar.tsx:88]: Oops, changed up wording and forgot to fix that

File: src/internal/puck/ui/puck.css [hunk 2]
```diff
@@ -44,7 +44,11 @@
 }
 
 .theme-field {
-  padding: 0;
+  color: var(--puck-color-grey-04);
```
[the remark this answers is not in the document]
Comment (asanehisa) [reply src/internal/puck/ui/puck.css:47]: None of the other stuff in the <AutoField/> component is stuff we want to attempt changing rn right?

[the remark this answers is not in the document]
Comment (mkilpatrick) [reply src/internal/puck/ui/puck.css:47]: Right. Your screenshot looks like what I was looking for.

**Review by mkilpatrick — APPROVED** [review]

**Review by benlife5 — APPROVED** [review]
🔥

**Pull request merged**